contribute/design/cstor-operator/specs.md

```diff
@@ -0,0 +1,235 @@
+the cStor storage pool name in the storage class to create the volume.
+
+For more information on how cStor works please check following link:
+https://docs.openebs.io/docs/next/cstor.html
+
+To provison a cStor pool, which is done by CStor-Operator, a set of kubernetes
+custom and native resources are created. The resources are:
+
+1. StoragePoolClaim(spc):
+
+   CStor-Operator watches for spc resource for creating pool resources i.e.
+   CStorPool custom resource and deployment which are describe below.
+
+2. CStorPool(csp):
+
+   CStor-Pool-Mgmt watches for csp resource to create cStor pool on node.
+
+3. CStor-Pool-Deployment:
+
+   A CStor pool deployment pod contains CStor-Pool-Mgmt container in addition to
+   other containers.
+
+In short, an OpenEBS user applies a StoragePoolClaim yaml to provision a cStor
+pool.
+
+For more information on CStor-Operator, please go through following link
+
+TODO: Put CStor-Operator design doc link.
+
+
+### Objectives
+
+This document will focus primarily on the spec of StoragePoolClaim and
+CStorPool custom resource.
+In addition to that, a proposal to rename StoragePoolClaim to CStorPoolCluster
+is brought which is discussed later in the section.
+
+Change in the specs of custom resources are brought to proposal to handle following:
+
+1. Handle basic day 2 operations.
+2. More granular control and visibilty on specs to user while provisioning pool.
+3. Enhanced reconciliation loop for the CStor-operator.
+
+### Proposals
+
+#### Rename StoragePoolClaim to CStorPoolCluster
+
+Generally, claim resources maps/bounds to a single object, for example PVC i.e.
+PersistentVolumeClaim bounds to a PV object. We can say, claim resources has
+a one to one mapping with some object.
+
+But StoragePoolClaim is mapped to a set of cStor pools which can create confusion
+and does not look intutive from users prespective.
+
+CStorPoolCluster looks intutive i.e. CStorPoolCluster is logically a resource
+which is parent of a collection of CStorPools.
+
+
+#### Spec for CStorPoolCluster
+
+Following are few sample manual CStorPoolCluster specs
+
+1.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-striped
+spec:
+  nodes:
+  - name: gke-cstor-it-default-pool-569eb31d-3l88
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      disks:
+        // Can use block device name here. In YAML spec 
+        // disks and block device will mean the same.
+      - name: sparse-3c1fc7491f9e4cf50053730740647318
+        id: disk1
+  name: cstor-pool-striped
+
+```
+
+2.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-striped
+spec:
+  nodes:
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      disks:
+      - name: sparse-pool-1-disk-1
+        id: pool-1-disk1
+      - name: sparse-pool-1-disk-2
+        id: pool-1-disk2
```

Comment:
there was requirement to provide 'replaced disk' as well right?

Reply:
As discussed -- Id will do the job.